File: #6 carrinho_de_compras/2018-1/elisson C#/Carrinho.cs
using System.Collections;
using System.Collections.Generic;

public class Carrinho {
	private List<Item> itens = new List<Item>();
	private float total;

	public void Adicionar(Produto produto){
		itens.Add (new Item(produto));
	}

	public List<Produto> Produtos() {
		List<Produto> produtos = new List<Produto>();
		foreach (Item item in this.itens) {
			produtos.Add (item.Produto);
		}
		return produtos;
	}

	public int Quantidade(Produto produto){
		Item item_com_produto = ItemContendoProduto (produto);
		return item_com_produto.Quantidade;
	}	

	public float Total() {
		foreach (Item item in this.itens) total += item.Produto.Preco;
		return total;
	}

	protected Item ItemContendoProduto(Produto produto){
		Item item_com_produto = null;
		foreach (Item item in this.itens) {
			if (item.Produto == produto) item_com_produto = item;
		}
		return item_com_produto;
	}
}

File: #6 carrinho_de_compras/2018-1/elisson java/Carrinho.java
import java.util.ArrayList;

public class Carrinho {
	private ArrayList<Item> itens;
	
	public Carrinho() {
		this.itens = new ArrayList<Item>();
	}
	
	public ArrayList<Produto> getProdutos() {
		ArrayList<Produto> produtos = new ArrayList<Produto>();
		for(Item item : itens) 	produtos.add(item.getProduto());
		return produtos;
	}
	
	public void incluir(Produto produto) {
		itens.add(new Item(produto));
	}
	
	public void remover(Produto produto) {
		itens.remove(itemComProduto(produto));
	}

	public void alterarQuantidade(Produto produto, int quantidade) {
		itemComProduto(produto).setQuantidade(quantidade);
	}
	
	public int getQuantidade(Produto produto) {
		return itemComProduto(produto).getQuantidade();
	}

	public float getTotal() {
		float total = 0;
		for(Item item : itens) total += item.valor();
		return total;
	}
	
	private Item itemComProduto(Produto produto) {
		for(Item item : itens) {
			if (item.getProduto() == produto) return item;
		}
		return null;
	}
}